### PR TITLE
Minor RectangleSelector refactors

### PR DESCRIPTION
**Pull request by dstansby**
I'm trying to get my head around some odd failures in https://github.com/matplotlib/matplotlib/pull/21945. As part of that process, here's some minor refactoring that improves readability of the RectangleSelector code and reduces a bit of duplication.

### CODE REVIEW

File: lib/matplotlib/widgets.py [hunk 1]
```diff
@@ -3026,26 +3039,11 @@ def _release(self, event):
...
-        minspanxy = (spanx <= self.minspanx or spany <= self.minspany)
-        if (self._drawtype != 'none' and minspanxy):
-            for artist in self.artists:
-                artist.set_visible(False)
+        if self._drawtype != 'none' and self._selector_too_small():
             if self._selection_completed:
                 # Call onselect, only when the selection is already existing
                 self.onselect(self._eventpress, self._eventrelease)
-            self._selection_completed = False
+            self.clear()
```
Comment (timhoffm) [inline lib/matplotlib/widgets.py:3046]: Is this change intended here? It's not a refactoring but changes code, e.g. `clear()` calls `update()` and then we call update again below in 3501.

Comment (ericpre) [reply lib/matplotlib/widgets.py:3046]: I don't remember what is the purpose of the `visible` attribute but another consequence of using `clear` is that it changes the value of the `visible` attribute from `True` to `False` and I am not sure if this desirable - this would need to be checked.

File: lib/matplotlib/widgets.py [hunk 2]
```diff
@@ -3005,6 +3005,19 @@ def _press(self, event):
 
         return False
 
+    def _selector_too_small(self):
```
Comment (ericpre) [inline lib/matplotlib/widgets.py:3008]: Should it be a property?

**Review by ericpre — COMMENTED** [review]
Since this code block is only used in a method which is not very long, I am not convinced that creating a separate method make the code more readable: this is about balancing redirection and adding more method versus shorten the code of where it is used, which in this case, I find subjective!

**Comment by dstansby** [comment]
Perhaps I was too hasty in opening this PR, and the changes make more sense in the context of https://github.com/matplotlib/matplotlib/pull/21945. I think I'll close this, fix up https://github.com/matplotlib/matplotlib/pull/21945 by itself, and apologise for the noise.

File: lib/matplotlib/widgets.py (hunk 1 above)
Comment (ericpre) [reply lib/matplotlib/widgets.py:3046]: For the record: it seems that the `visible` attribute is only used by the selector itself (and should be private) and it is reset in `_press`, so the point I made here is not an issue!

**Comment by ericpre** [comment]
I fully appreciate that the selector are not easy to read and I am really looking forward for when we can remove the `drawtype` argument and all the clean up that goes with it! :)

**Comment by dstansby** [comment]
😆 couldn't agree more!
